sdk/core/core-http/lib/util/xml.ts

```diff
@@ -35,3 +38,152 @@ export function parseXML(str: string): Promise<any> {
     }
   });
 }
+
+export function deserializeAtomXmlToJson(body: string): any {
+  let parsed;
+  const parser = new xml2js.Parser(_getDefaultSettingsForAtomXmlOperations());
+  parser.parseString(_removeBOM(body.toString()), function(err: any, parsedBody: any) {
```

Comment:
Is `parseString` non-blocking? Curious because it accepts a callback function.

Reply:
It seems blocking, because if it weren't then parsed should have been returning null/defined always?